test/performance/odp_sched_pktio.c

```diff
@@ -1047,21 +1054,74 @@ static void link_pktios(test_global_t *test_global)
 	printf("\n");
 }
 
+static int print_link_info(odp_pktio_t pktio)
+{
+	odp_pktio_link_info_t info;
+
+	if (odp_pktio_link_info(pktio, &info)) {
+		ODPH_ERR("Error: Pktio link info failed.\n");
+		return -1;
+	}
+
+	printf("  autoneg     %s\n",
+	       (info.autoneg == ODP_PKTIO_LINK_AUTONEG_ON ? "on" :
+	       (info.autoneg == ODP_PKTIO_LINK_AUTONEG_OFF ? "off" : "unknown")));
+	printf("  duplex      %s\n",
+	       (info.duplex == ODP_PKTIO_LINK_DUPLEX_HALF ? "half" :
+	       (info.duplex == ODP_PKTIO_LINK_DUPLEX_FULL ? "full" : "unknown")));
+	printf("  media       %s\n", info.media);
+	printf("  pause_rx    %s\n",
+	       (info.pause_rx == ODP_PKTIO_LINK_PAUSE_ON ? "on" :
+	       (info.pause_rx == ODP_PKTIO_LINK_PAUSE_OFF ? "off" : "unknown")));
+	printf("  pause_tx    %s\n",
+	       (info.pause_tx == ODP_PKTIO_LINK_PAUSE_ON ? "on" :
+	       (info.pause_tx == ODP_PKTIO_LINK_PAUSE_OFF ? "off" : "unknown")));
+	printf("  speed(Mbit/s) %" PRIu32 "\n\n", info.speed);
+
+	return 0;
+}
+
 static int start_pktios(test_global_t *test_global)
 {
-	int i;
+	uint32_t i;
+	test_options_t *test_options = &test_global->opt;
+	uint32_t num_pktio = test_options->num_pktio;
+	uint32_t link_wait = 0;
 
-	for (i = 0; i < test_global->opt.num_pktio; i++) {
+	for (i = 0; i < num_pktio; i++) {
 		if (odp_pktio_start(test_global->pktio[i].pktio)) {
-			printf("Error (%s): Pktio start failed.\n",
-			       test_global->opt.pktio_name[i]);
+			ODPH_ERR("Error (%s): Pktio start failed.\n", test_options->pktio_name[i]);
 
 			return -1;
 		}
 
 		test_global->pktio[i].started = 1;
 	}
 
+	/* Wait until all links are up */
+	for (i = 0; test_options->wait_sec && i < num_pktio; i++) {
+		while (1) {
```

Comment:
You could utilize `link_wait` here, i.e., initialize with `test_options->wait_sec` and then `while (link_wait--)`.